wire-runtime-swift/src/test/swift/ProtoWriterTests.swift

```diff
@@ -255,6 +255,56 @@ final class ProtoWriterTests: XCTestCase {
         """)
     }
 
+    func testEncodeMessageWithStackExpansion() throws {
+        let writer = ProtoWriter()
+        let message = Nested1(name: "name1") { v1 in
+          v1.nested = Nested2(name: "name2") { v2 in
+            v2.nested = Nested3(name: "name3") { v3 in
+              v3.nested = Nested4(name: "name4") { v4 in
+                v4.nested = Nested5(name: "name5") { v5 in
+                  v5.nested = Nested6(name: "name6")
+                }
+              }
+            }
+          }
+        }
+
+        try writer.encode(tag: 1, value: message)
+
+        assertBufferEqual(writer, """
```

Comment:
this test fails without the change.

here's the bytes after the fix, followed by the bytes before the fix. before the fix one of the bytes is actually missing - the `0A (Tag 1 | Length Delimited)` byte that preceeds the `name2` string value

      // 0A 34 0A 05 6E616D6531 12 2B 0A 05 6E616D6532 12 22 0A 05 6E616D6533 12 19 0A 05 6E616D65 3412 10 0A 05 6E616D6535 1207 0A 05 6E616D6536
      // 0A 34 0A 05 6E616D6531 12 2B XX 05 6E616D6532 12 22 0A 05 6E616D6533 12 19 0A 05 6E616D65 3412 10 0A 05 6E616D6535 1207 0A 05 6E616D6536
      //                               \
      //                                \_ missing byte

Reply:
this doesn't seem to repro for me... did you have to do anything special? what toolchain were you using?

Reply:
I was using Xcode 16.0

Reply:
what part doesn't repro for you?

Reply:
Hmm, I can't seem to repro the corrupted data either. Perhaps I encountered this while testing some other fix? I'm not sure. I was copying those hex strings from the assertions that were failing 🤔

Reply:
Assuming this was a false signal I encountered then its a relief that the actual data that was encoded was not impacted

Reply:
> what part doesn't repro for you?

the test still passes even if i run with the implementation change as it was vs the fixed version.

> Assuming this was a false signal I encountered then its a relief that the actual data that was encoded was not impacted

agreed. but does the test actually fail then without the change? obviously good to have it regardless so that something is exercising the 'grow the buffer' path, but also makes me feel it might provide less safety than it might aspire to. ideally these tests should probably run in a mode with ASAN on as part of CI.